Sources/NIOSSL/SwiftCrypto/NIOSSLSecureBytes.swift

```diff
@@ -0,0 +1,429 @@
+        self.backing = Backing.create(capacity: unsafeUninitializedCapacity)
+        try self.backing._withVeryUnsafeMutableBytes { veryUnsafePointer in
+            // As Array does, we want to truncate the initializing pointer to only have the requested size.
+            var veryUnsafePointer = UnsafeMutableRawBufferPointer(rebasing: veryUnsafePointer.prefix(unsafeUninitializedCapacity))
+            var initializedCount = 0
+            try callback(&veryUnsafePointer, &initializedCount)
+
+            self.backing.count = initializedCount
+        }
+    }
+}
+
+extension NIOSSLSecureBytes {
+    @inlinable
+    mutating func append<C: Collection>(_ data: C) where C.Element == UInt8 {
+        let requiredCapacity = self.count + data.count
+        if !isKnownUniquelyReferenced(&self.backing) || requiredCapacity > self.backing.capacity {
+            let newBacking = Backing.create(capacity: requiredCapacity)
+            newBacking._appendBytes(self.backing, inRange: 0..<self.count)
+            self.backing = newBacking
+        }
+        self.backing._appendBytes(data)
+    }
+
+    
+    mutating public func reserveCapacity(_ n: Int) {
+        if self.backing.capacity >= n {
+            return
+        }
+
+        let newBacking = Backing.create(capacity: n)
+        newBacking._appendBytes(self.backing, inRange: 0..<self.count)
+        self.backing = newBacking
+    }
+}
+
+// MARK: - Equatable conformance, constant-time
+extension NIOSSLSecureBytes: Equatable {
+    public static func == (lhs: NIOSSLSecureBytes, rhs: NIOSSLSecureBytes) -> Bool {
+        //return safeCompare(lhs, rhs)
+        lhs.backing.withUnsafeBytes { lhsPtr in
+            rhs.backing.withUnsafeBytes { rhsPtr in
+                constantTimeCompare(lhsPtr, rhsPtr)
+            }
+        }
+    }
+}
+
+// MARK: - Collection conformance
+extension NIOSSLSecureBytes: Collection {
+    
+    public struct Index {
+        /* fileprivate but usableFromInline */ @usableFromInline var offset: Int
+
+        /*@inlinable*/ @usableFromInline internal init(offset: Int) {
+            self.offset = offset
+        }
+    }
+
+    @inlinable
+    public var startIndex: Index {
+        return Index(offset: 0)
+    }
+
+    @inlinable
+    public var endIndex: Index {
+        return Index(offset: self.count)
+    }
+
+    @inlinable
+    public var count: Int {
+        return self.backing.count
+    }
+
+    @inlinable
+    public subscript(_ index: Index) -> UInt8 {
+        get {
+            return self.backing[offset: index.offset]
+        }
+        set {
+            self.backing[offset: index.offset] = newValue
+        }
+    }
+
+    @inlinable
+    public func index(after index: Index) -> Index {
+        return index.advanced(by: 1)
+    }
+    
+    @inlinable
+    public var regions: CollectionOfOne<NIOSSLSecureBytes> {
```

Comment:
We don't need this: this is part of the `DataProtocol` conformance we don't have anymore.

Reply:
Done in 8007b79